cookbook/B66_gadget.rst

```diff
@@ -0,0 +1,534 @@
+
+
+.. code-block:: yaml
+
+  - time: 0
+    lights: 
+      (gadget_letter_made_led): (gadget_letter_made_color)
+
+  - time: +.05
+    lights: 
+      (gadget_letter_made_led): off
+
+  - time: +.05
+    lights: 
+      (gadget_letter_made_led): (gadget_letter_made_color)
+
+  - time: +.05
+    lights: 
+      (gadget_letter_made_led): off
+
+  - time: +.05
+    lights: 
+      (gadget_letter_made_led): (gadget_letter_made_color)
+
+  - time: +.05
+    lights: 
+      (gadget_letter_made_led): off
+
+  - time: +.05
+    lights: 
+      (gadget_letter_made_led): (gadget_letter_made_color)
+
+  - time: +.05
+    lights: 
+      (gadget_letter_made_led): off
+
+  - time: +.05
+    lights: 
+      (gadget_letter_made_led): (gadget_letter_made_color)
+
+  - time: +.05
+    lights: 
+      (gadget_letter_made_led): off
+
+  - time: +.05
+    lights: 
+      (gadget_letter_made_led): (gadget_letter_made_color)
+
+  - time: +.05
+    lights: 
+      (gadget_letter_made_led): off
+
+  - time: +.05
+    lights: 
+      (gadget_letter_made_led): (gadget_letter_made_color)
+
+  - time: +.05
+    lights: 
+      (gadget_letter_made_led): off
+
+  - time: +.05
+    lights: 
+      (gadget_letter_made_led): (gadget_letter_made_color)
+
+  - time: +.05
+    lights: 
+      (gadget_letter_made_led): off
+
+  - time: +.05
+    lights: 
+      (gadget_letter_made_led): (gadget_letter_final_color)
+
+    duration: -1
+
+
+
+This show isn't terribly complicated, but let's look at some of the features.
+
+.. code-block:: yaml
+
+  - time: 0
+    lights: 
+      (gadget_letter_made_led): (gadget_letter_made_color)
+
+  - time: +.05
+    lights: 
+      (gadget_letter_made_led): off
+
+When the show starts, it accepts a token from the ``show_player`` (we'll configure that next), that tells MPF what corresponding light(s) we're going to flash, and what color to flash them. 
+
+In a real Batman '66, we would simply flash the light because the inserts are yellow. However, since many custom games are using RGB LED, we'll allow for any color the builder prefers.
+
+
+.. code-block:: yaml
+
+  - time: +.05
+    lights: 
+      (gadget_letter_made_led): (gadget_letter_final_color)
+
+    duration: -1
+
+The last step is special for two reasons. We're passing in a second color that will be 'held' at the end of the show indefinitely as indicated by ``duration -1``. We've done this in order to allow for the same show to end in a 'lit' or 'unlit' state, depending on our need in a situation.
+
+
+
+In the code you can download from the link at the beginning of this cookbook, there is another show that lights the LED on the lockdown bar, but it's not worth explaining here.
+
+Step 6. Configure the Show Player
+---------------------------------
+
+Our show player is watching for events and triggering the appropriate shows.
+
+.. code-block:: yaml
+
+  show_player:
+    gadget_g1_complete:
```

Comment:
How does this handle if the mode stops and restarts? Is the expectation that all the letters reset on each ball, so we never have to re-set the show to its previous state?

Reply:
The game carries the state of letters between balls. i.e. If I have D E  when I drain ball 1, they remain lit when ball 2 starts.